contracts/src/strealm.cairo

```diff
@@ -0,0 +1,294 @@
+        +ERC721Component::ERC721HooksTrait<TContractState>,
+        +SRC5Component::HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of super::IStRealm<ComponentState<TContractState>> {
+        //
+        // Immutables
+        //
+        fn get_stream(self: @ComponentState<TContractState>, owner: ContractAddress) -> Stream {
+            self.StRealm_streams.read(owner)
+        }
+
+        fn get_latest_flow(self: @ComponentState<TContractState>) -> Flow {
+            let latest_flow_id = self.StRealm_latest_flow_id.read();
+            self.StRealm_flows.read(latest_flow_id)
+        }
+
+        //
+        // Mutables
+        //
+        fn claim(ref self: ComponentState<TContractState>) {
+            self._claim_stream(starknet::get_caller_address())
+        }
+
+        fn update_flow_rate(ref self: ComponentState<TContractState>, new_rate: u256) {
+            let accesscontrol_component = get_dep_component!(@self, AccessControl);
+            accesscontrol_component.assert_only_role(DEFAULT_ADMIN_ROLE);
+
+            self._update_flow_rate(new_rate);
+        }
+
+        fn update_reward_token(
+            ref self: ComponentState<TContractState>, new_token_address: ContractAddress
+        ) {
+            let accesscontrol_component = get_dep_component!(@self, AccessControl);
+            accesscontrol_component.assert_only_role(DEFAULT_ADMIN_ROLE);
+
+            self._update_reward_token(new_token_address);
+        }
+
+        fn update_reward_payer(
+            ref self: ComponentState<TContractState>, new_payer_address: ContractAddress
+        ) {
+            let accesscontrol_component = get_dep_component!(@self, AccessControl);
+            accesscontrol_component.assert_only_role(DEFAULT_ADMIN_ROLE);
+
+            self._update_reward_payer(new_payer_address);
+        }
+    }
+
+
+    //
+    // Internal
+    //
+
+    #[generate_trait]
+    impl InternalImpl<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl ERC721: ERC721Component::HasComponent<TContractState>,
+        impl SRC5: SRC5Component::HasComponent<TContractState>,
+        +ERC721Component::ERC721HooksTrait<TContractState>,
+        +Drop<TContractState>,
+    > of InternalTrait<TContractState> {
+        /// This should be used inside the contract's constructor.
+        fn initializer(
+            ref self: ComponentState<TContractState>,
+            flow_rate: u256,
+            reward_token: ContractAddress,
+            reward_payer: ContractAddress
+        ) {
+            self._update_flow_rate(flow_rate);
+            self._update_reward_token(reward_token);
+            self._update_reward_payer(reward_payer);
+        }
+
+
+        fn _update_flow_rate(ref self: ComponentState<TContractState>, new_flow_rate: u256) {
+            self._end_latest_flow();
+            self._start_new_flow(new_flow_rate);
+        }
+
+        fn _update_reward_token(
```

Comment:
the reward token should only ever be lords. i'll remove the update reward token funtion

Reply:
Got it.

Yeah, you could remove the function and also the `RewardTokenUpdated` event (as it would only ever be triggered once), probably would result in slightly smaller code to declare. This way is also fine though. Your call, resolving the convo.